sbncode/TPCReco/VertexStub/StubMergeAlgorithms.cxx

```diff
@@ -149,46 +137,48 @@ float sbn::StubDirectionDot(const sbn::StubInfo &A, const sbn::StubInfo &B,
+geo::Point_t sbn::TwoStubEndPosition(const sbn::StubInfo &A, const sbn::StubInfo &B,
     const geo::GeometryCore *geo,
     const spacecharge::SpaceCharge *sce,
     const detinfo::DetectorPropertiesData &dprop) {
 
   static constexpr bool VERBOSE = false;
 
-  // look up the wire-coordinate of both wires
-  double A_w = T.WireCoordinate(geo, A.vhit_hit->WireID());
-  double B_w = T.WireCoordinate(geo, B.vhit_hit->WireID());
-
-  // Use this to get y, z
-  double y = T.TwoPlaneToY(geo->View(A.vhit_hit->WireID()), A_w, geo->View(B.vhit_hit->WireID()), B_w);
-  double z = T.TwoPlaneToZ(geo->View(A.vhit_hit->WireID()), A_w, geo->View(B.vhit_hit->WireID()), B_w);
+  // intersect the wires with the geometry service
+  geo::Point_t yz;
+  bool intersects = geo->WireIDsIntersect(A.vhit_hit->WireID(), B.vhit_hit->WireID(), yz);
+  (void) intersects; // TODO: Do we care if the wires intersect? Means the end point will be outside the AV
+  double y = yz.Y();
+  double z = yz.Z();
 
   // just average the x-pos
   double x = (dprop.ConvertTicksToX(A.vhit_hit->PeakTime(), A.vhit_hit->WireID()) + dprop.ConvertTicksToX(B.vhit_hit->PeakTime(), B.vhit_hit->WireID())) / 2.;
 
   geo::Point_t pos(x, y, z);
 
   // Correct for space charge
-  pos = GetLocation(sce, pos, A.vhit_hit->WireID().TPC);
+  pos = GetLocation(sce, pos, A.vhit_hit->WireID());
 
   if (VERBOSE) {
     std::cout << "A View: " << geo->View(A.vhit_hit->WireID()) << " Angle to Vertical: " << geo->WireAngleToVertical(geo->View(A.vhit_hit->WireID()), A.vhit_hit->WireID()) << std::endl;
     std::cout << "B View: " << geo->View(B.vhit_hit->WireID()) << " Angle to Vertical: " << geo->WireAngleToVertical(geo->View(B.vhit_hit->WireID()), B.vhit_hit->WireID()) << std::endl;
+    std::cout << "A End: " << A.stub.end.X() << " " << A.stub.end.Y() << " " << A.stub.end.Z() << std::endl;
+    std::cout << "B End: " << B.stub.end.X() << " " << B.stub.end.Y() << " " << B.stub.end.Z() << std::endl;
     std::cout << "A Wire Pos Y: " << geo->Wire(A.vhit_hit->WireID()).GetCenter().Y() << " Z: " << geo->Wire(A.vhit_hit->WireID()).GetCenter().Z() << std::endl;
     std::cout << "B Wire Pos Y: " << geo->Wire(B.vhit_hit->WireID()).GetCenter().Y() << " Z: " << geo->Wire(B.vhit_hit->WireID()).GetCenter().Z() << std::endl;
     
-    std::cout << "A Wire: " << A.vhit_hit->WireID().Wire << " Plane: " << A.vhit_hit->WireID().Plane << " TPC: " << A.vhit_hit->WireID().TPC << " Coord: " << A_w << std::endl;
-    std::cout << "B Wire: " << B.vhit_hit->WireID().Wire << " Plane: " << B.vhit_hit->WireID().Plane << " TPC: " << B.vhit_hit->WireID().TPC << " Coord: " << B_w << std::endl;
+    std::cout << "A Wire: " << A.vhit_hit->WireID().Wire << " Plane: " << A.vhit_hit->WireID().Plane << " TPC: " << A.vhit_hit->WireID().TPC << std::endl;
+    std::cout << "B Wire: " << B.vhit_hit->WireID().Wire << " Plane: " << B.vhit_hit->WireID().Plane << " TPC: " << B.vhit_hit->WireID().TPC << std::endl;
```

Comment:
You can also write a `geo::WireID` directly:
```suggestion
    std::cout << "A Wire: " << A.vhit_hit->WireID() << std::endl;
    std::cout << "B Wire: " << B.vhit_hit->WireID() << std::endl;
````
It print something like: `C:0 T:1 P:2 W:345`.